Doctest failing

The doctests are currently failing across branches. The error message is
```
Exception occurred:
  File "<template>", line 3, in top-level template code
TypeError: can only concatenate str (not "_StrPath") to str
The full traceback has been saved in /tmp/sphinx-err-b73gufkd.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
nox > Command sphinx-build -b html -W --keep-going . /home/runner/work/PyBaMM/PyBaMM/.nox/docs/bin/../tmp/html failed with exit code 2
nox > Session docs failed.
Error: Process completed with exit code 1.
```
but for a full log please see: https://github.com/pybamm-team/PyBaMM/actions/runs/11208039171/job/31151139329

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -341,7 +341,7 @@ nbsphinx_prolog = r"""
 
 {% set github_docname =
 'github/pybamm-team/pybamm/blob/develop/docs/' +
-str(env.doc2path(env.docname, base=None)) %}
+env.doc2path(env.docname, base=None) | string() %}
 
 {% set notebooks_version = env.config.html_context.notebooks_version %}
 {% set github_download_url = env.config.html_context.github_download_url %}

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -341,13 +341,13 @@ nbsphinx_prolog = r"""
 
 {% set github_docname =
 'github/pybamm-team/pybamm/blob/develop/docs/' +
-env.doc2path(env.docname, base=None) %}
+str(env.doc2path(env.docname, base=None)) %}
 
 {% set notebooks_version = env.config.html_context.notebooks_version %}
 {% set github_download_url = env.config.html_context.github_download_url %}
 {% set google_colab_url = env.config.html_context.google_colab_url %}
 
-{% set doc_path = env.doc2path(env.docname, base=None) %}
+{% set doc_path = str(env.doc2path(env.docname, base=None)) %}
 
 .. raw:: html
 

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -347,7 +347,7 @@ str(env.doc2path(env.docname, base=None)) %}
 {% set github_download_url = env.config.html_context.github_download_url %}
 {% set google_colab_url = env.config.html_context.google_colab_url %}
 
-{% set doc_path = str(env.doc2path(env.docname, base=None)) %}
+{% set doc_path = env.doc2path(env.docname, base=None) %}
 
 .. raw:: html
 

